Makefile

```diff
@@ -94,6 +94,13 @@ staticcheck:
 			  staticcheck $(PKGS); \
 			  staticcheck -tags integrationtest test/integration_test/*.go;staticcheck -tags unittest $(PKGS)"
 
+mockgen:
```

Comment:
Does mockgen need to be run as part of `make test`?

Reply:
Not really. We just need to run it if we need to update the mock files.

Reply:
I see your point. If the original interfaces change when they get updated in the vendor directory, it is better to generate the mocks again. I will change it in the other PR I have.